app/src/main/java/cc/pogoda/mobile/meteosystem/Main.java

```diff
@@ -125,10 +126,9 @@ public void onCreate() {
 
         favouritiesFile = new FavouritiesFile(fileNames);
 
-        // download all stations from API
-        listOfAllStations = new AllStationsDao().getAllStations();
-
-        Logger.info("[Main][onCreate][listOfAllStations.size() = " + listOfAllStations.size() +  "]");
+        // Download all stations from API in background via JobIntentService. Results are send
+        //back with Broadcast receiver.
+        startGetAllStationsService();
 
         // recreate list of favorites
         recreateListOfFavs();
```

Comment:
Ah and one little caveat about line 136 and 138. It is worth considering if this updater should be created (and/or started) here or inside the service after the station list will be available. The idea behind the updater is that it should do it's job before a user will click 'Favorites' button, to minimize or preferably eliminate at all a delay with which measurements summary will be displayed below each entry. To minimize API load (and GPRS transfer etc) task is scheduled by an executor to 123 seconds. If it will be missed here (I mean where I put this comment) a user won't see this  
```
    public void start(int _initial_delay) {

        Logger.debug("[FavouritesStationSummaryUpdater][start][_initial_delay = " + _initial_delay +"]");

        if (enabled) {
            stop();
        }

        scheduledTask = executor.scheduleAtFixedRate(this, _initial_delay, 123000, TimeUnit.MILLISECONDS);
        enabled = true;
    }
```